### PR TITLE
Issue 992 plot arrays

### LINKED ISSUE #992: plot pybamm arrays
It would be nice to automate the plotting of `pybamm.Array` objects for doing things like the example below. I think just 1D is fine for now. More complicated will usually be of solution entries, so handled via `Quickplot`, but it would be nice to easily plot parameters that are functions. I guess 2D is useful since this depends on T and sto. Thoughts?

Edit: I know calling `.entries` isn't a huge deal, but think this would be easy to implement and avoid needing to import matplotlib 

```
import pybamm
import numpy as np
import matplotlib.pyplot as plt

param = pybamm.ParameterValues(chemistry=pybamm.parameter_sets.Ecker2015)
D_fun = param["Negative electrode diffusivity [m2.s-1]"]
x = pybamm.Array(np.linspace(0,1,100))
D = D_fun(x, 300)
plt.plot(x.entries,D.entries) 
```

### PR DESCRIPTION
**Pull request by rtimms**
# Description
Added basic plots for 1D and 2D arrays (`pybamm.plot` and `pybamm.plot2D`). I've left these as separate methods, but I guess they could be combined and the appropriate plot called depending on what arrays you pass. On the other hand, maybe it is better to be explicit. 

Also added `pybamm.linspace` and `pybamm.meshgrid` to quickly create some common arrays.

Fixes #992 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/master/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/__init__.py [hunk 1]
```diff
@@ -223,7 +223,7 @@ def version(formatted=False):
 #
 # other
 #
-from .quick_plot import QuickPlot, dynamic_plot, ax_min, ax_max
+from .quick_plot import QuickPlot, plot, plot2D, dynamic_plot, ax_min, ax_max
```
Comment (tinosulzer) [inline pybamm/__init__.py:226]: might be worth making a new plotting folder with a few different functions in different files?

File: tests/unit/test_plot.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            pybamm.plot("bad", x)
+        with self.assertRaisesRegex(TypeError, "y must be 'pybamm.Array'"):
+            pybamm.plot(x, "bad")
+
+    def test_plot2D(self):
+        x = pybamm.Array(np.array([0, 3, 10]))
+        y = pybamm.Array(np.array([6, 16, 78]))
+        X, Y = pybamm.meshgrid(x, y)
+
+        pybamm.plot2D(X, Y, Y, xlabel="x", ylabel="y", title="title", testing=True)
```
Comment (tinosulzer) [inline tests/unit/test_plot.py:24]: could `plot2D` automatically do meshgrid(x,y) if they aren't given in meshgrid form?

File: pybamm/expression_tree/array.py [hunk 3]
```diff
@@ -101,3 +101,16 @@ def new_copy(self):
     def _base_evaluate(self, t=None, y=None, y_dot=None, inputs=None):
         """ See :meth:`pybamm.Symbol._base_evaluate()`. """
         return self._entries
+
+
+def linspace(start, stop, num=50, **kwargs):
+    "Creates a linearly spaced array"
```
Comment (tinosulzer) [inline pybamm/expression_tree/array.py:107]: why pick out `num` from the `kwargs`? Also, link to `np.linspace` docs for the kwargs

File: pybamm/expression_tree/array.py [hunk 4]
```diff
@@ -101,3 +101,16 @@ def new_copy(self):
...
+    return pybamm.Array(np.linspace(start, stop, num, **kwargs))
+
+
+def meshgrid(x, y, **kwargs):
+    "Return coordinate matrices as from coordinate vectors"
```
Comment (tinosulzer) [inline pybamm/expression_tree/array.py:112]: link `np.meshgrid` docs for the kwargs

File: tests/unit/test_plot.py (hunk 2 above)
Comment (rtimms) [reply tests/unit/test_plot.py:28]: turns out contourf handles this provided they are the correct shape. have added some lines to account for this

File: pybamm/__init__.py (hunk 1 above)
Comment (rtimms) [reply pybamm/__init__.py]: agreed

File: pybamm/expression_tree/array.py (hunk 3 above)
Comment (rtimms) [reply pybamm/expression_tree/array.py]: did this so you can do `pybamm.linspace(0,1,100)` without needing to do `num=100` as I think it's a pretty common use case and you can just do `np.linspace(0,1,100)` directly. that didn't work unless I picked out `num` but maybe there is a better/more common solution?

**Pull request merged**